Azure.MgmtSdk.Analyzers/ModelNameSuffixCondition.cs

```diff
@@ -0,0 +1,168 @@
+﻿using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.Diagnostics;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Text.RegularExpressions;
+using System;
+using System.ComponentModel;
+
+namespace Azure.MgmtSdk.Analyzers
+{
+    /// <summary>
+    /// Analyzer to check type name suffixes. There are some suffixed we don't recommend to use, like `Result`, `Response`...
+    /// </summary>
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class ModelNameSuffixCondition : DiagnosticAnalyzer
```

Comment:
Better to name after `XXXXAnalyzer` so that it can be easily differentiated from other domain classes.